### PR TITLE
#3679 remove duplicate temperature breach methods

### LINKED ISSUE #3697: Remove duplicated TemperatureBreach methods
## The suggestion

Remove unused `TemperatureBreach` methods:

- `willContinueBreach(temperatureLog)`
- `get colour()`

**Why?**

Unused and non-functional (will throw errors if called).

## Describe alternatives you've considered

N/A.

## Additional context

N/A.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #3697.

## Change summary

Just tidying up by removing some old methods. 

## Testing

Steps to reproduce or otherwise test the changes of this PR:

N/A.

### Related areas to think about

N/A.

### CODE REVIEW

File: src/database/DataTypes/TemperatureBreach.js [hunk 1]
```diff
@@ -7,10 +7,6 @@ import Realm from 'realm';
 import moment from 'moment';
 
 export class TemperatureBreach extends Realm.Object {
-  get colour() {
```
Comment (katherine-sussol) [inline src/database/DataTypes/TemperatureBreach.js:10]: Just double checking if this line is a potential reference?  https://github.com/openmsupply/mobile/blob/c21dedf5f31bd667ab284dff76608adfc8241b2f/src/widgets/BreachChart.js#L22

`colour` doesn't ring a bell to me though and looks like it's usually `null`...unless configured in desktop somewhere 🤔 ?

File: src/database/DataTypes/TemperatureBreach.js [hunk 2]
```diff
@@ -76,13 +72,6 @@ export class TemperatureBreach extends Realm.Object {
     return incomingStock.sum('numberOfPacks') ?? 0 - outgoingStock.sum('numberOfPacks') ?? 0;
   }
 
-  willContinueBreach(temperatureLog) {
```
Comment (katherine-sussol) [inline src/database/DataTypes/TemperatureBreach.js:79]: 👍

File: src/database/DataTypes/TemperatureBreach.js (hunk 1 above)
Comment (mark-prins) [reply src/database/DataTypes/TemperatureBreach.js:10]: it is indeed.
@wlthomson would you care to comment? Would be nice to wrap up this PR

Comment (wlthomson) [reply src/database/DataTypes/TemperatureBreach.js:10]: Well `this.temperatureBreachConfiguration?.colour ?? null` is always going to be `undefined` (as `temperatureBreachConfiguration` is not a field. I might be missing something, but not sure how desktop config would make a difference? Are you suggesting just replace the method with `return null`? In that case, is there much advantage of doing that over just leaving it without the method and returning `undefined`?

Comment (katherine-sussol) [reply src/database/DataTypes/TemperatureBreach.js:10]: I see, that's a fair point, didn't see that `this.temperatureBreachConfiguration` wasn't even a field on `TemperatureBreach`. I was concerned there was a use case where a colour could be set up in Desktop and there was an expectation of colour changing in `BreachChart`, but now I see that it wouldn't work regardless of this PR haha. All good then 😆 .

**Review by katherine-sussol — APPROVED** [review]
👍

File: src/database/DataTypes/TemperatureBreach.js (hunk 1 above)
Comment (wlthomson) [reply src/database/DataTypes/TemperatureBreach.js:10]: I'm not sure, it looks like the changes were introduced here: https://github.com/openmsupply/mobile/pull/3302/files.

@joshxg any thoughts?

Comment (joshxg) [reply src/database/DataTypes/TemperatureBreach.js:10]: Yeah there was intention to use colour on the config to colour the breach chart. Been scrapped.

The config relation on a breach was also changed to explicitly saved the values instead for ensuring history was kept.

Can be removed. It is also referenced elsewhere (breach chart) which could be removed

**Comment by wlthomson** [comment]
Have removed the `colour` logic from `BreachChart`, might be good to get a quick re-review just in case (sorry!).

**Comment by katherine-sussol** [comment]
Hmm ok, I think I mentally skipped this PR for ages, oops - latest commit looks good, should probs just merge lol.

**Pull request merged**